### ISSUE TITLE
Ignore TypeScript type aliases in space-infix-ops

### ISSUE THREAD

**Issue by madbence** [eslint/eslint#10922]
**What rule do you want to change?**

`space-infix-ops`

**Does this change cause the rule to produce more or fewer warnings?**

No change for javascript, no more false positives for typescript.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Disable the rule for `VariableDeclarator`s where the parent node is a `VariableDeclaration` with a non-standard `kind` (not `var`, `let`, or `const`). After this, [`eslint-plugin-typescript`](https://github.com/nzakas/eslint-plugin-typescript) can implement the rule for typescript type aliases [(which are represented as variable declarations, with `kind: 'type'`)](https://github.com/JamesHenry/typescript-estree/blob/c165bd334751fcd954c6f699d15647c97e785794/lib/convert.js#L2111-L2146).

**Please provide some example code that this change will affect:**

```js
type Foo<T> = T;
```

**What does the rule currently do for this code?**

It reports a violation, as after `Foo` the `<` is not surrounded by whitespaces.

**What will the rule do after it's changed?**

It'll ignore typescript type aliases, so a plugin can handle the non-standard syntax.

**Comment by kaicataldo** [comment]
Thanks for this issue. Since ESLint does officially support type annotations (as defined by [ESTree](https://github.com/estree/estree/blob/master/extensions/type-annotations.md)), I wonder if we can add an option that makes `space-infix-ops` ignore types? If this doesn't cover all specific TS syntax, it might be best to create a new `space-infix-ops` specifically for TS (maybe in https://github.com/nzakas/eslint-plugin-typescript)?

**Comment by madbence** [comment]
@kaicataldo i refactored the original rule, and eliminated the false positives. see #10935
